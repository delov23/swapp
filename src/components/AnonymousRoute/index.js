export { default as AnonymousRoute } from './AnonymousRoute';
