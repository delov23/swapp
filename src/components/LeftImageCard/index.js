export { default as LeftImageCard } from './LeftImageCard';
