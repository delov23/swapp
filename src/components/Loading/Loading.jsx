import React, { useContext } from 'react';

import styles from './Loading.module.css';
import { themify } from '../../utils/themify';
import ThemeContext from '../../context/ThemeContext';

const Loading = () => {
  const theme = useContext(ThemeContext);

  return (
    <div className={styles.wrapper}>
      <h1 className={styles[themify('loader', theme)]}>SW</h1>
    </div>
  );
};

export default Loading;
