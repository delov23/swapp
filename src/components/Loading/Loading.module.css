.wrapper {
  min-height: inherit;
  display: flex;
  justify-content: center;
  align-items: center;
}

.loader {
  font-family: Starwars;
  text-align: center;
  font-size: 10vh;
  animation: loading 1.5s infinite;
  -webkit-animation: loading 1.5s infinite;
}

.loaderLight {
  composes: loader;
  color: var(--mainYellow);
}

.loaderDark {
  composes: loader;
  color: var(--mainCyan);
}

@keyframes loading {
  0% {
    transform: scale(1);
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    -o-transform: scale(1);
  }
  50% {
    transform: scale(1.4);
    -webkit-transform: scale(1.4);
    -moz-transform: scale(1.4);
    -ms-transform: scale(1.4);
    -o-transform: scale(1.4);
  }
  100% {
    transform: scale(1);
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    -o-transform: scale(1);
  }
}
