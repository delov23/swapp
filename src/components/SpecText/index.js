export { default as SpecText } from './SpecText';
