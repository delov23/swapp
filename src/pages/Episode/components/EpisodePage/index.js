export { default as EpisodePage } from './EpisodePage';
