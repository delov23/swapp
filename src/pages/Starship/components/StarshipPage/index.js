export { default as StarshipPage } from './StarshipPage';
