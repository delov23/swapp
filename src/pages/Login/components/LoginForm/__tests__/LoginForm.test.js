import React from 'react';
import { shallow, mount } from 'enzyme';

import { LoginForm } from '../';
import { Loading } from '../../../../../components/Loading';
import { ApolloError } from 'apollo-client';

describe('<LoginForm />', () => {
  it('should match snapshot', () => {
    expect(shallow(<LoginForm />)).toMatchSnapshot();
  });

  it('should submit', () => {
    const handleSubmit = jest
      .fn()
      .mockImplementation(cb => () => cb({ test: 'test' }));
    const wrapper = mount(<LoginForm handleLogin={handleSubmit} />);

    wrapper.find('form').simulate('submit');
    expect(handleSubmit).toBeCalledTimes(1);
  });

  it('should change the input', async () => {
    const handleSubmit = jest
      .fn()
      .mockImplementation(cb => () => cb({ test: 'test' }));
    const wrapper = mount(<LoginForm handleLogin={handleSubmit} />);

    wrapper
      .find('input[name="email"]')
      .simulate('change', { target: { value: 'val', name: 'email' } });

    expect(wrapper.find('input[name="email"]')).toHaveValue('val');
  });

  it('should display loading', () => {
    const wrapper = mount(<LoginForm loading={true} />);

    expect(wrapper.find(Loading)).toHaveLength(1);
  });

  it('should display error', () => {
    const wrapper = mount(
      <LoginForm error={new ApolloError({ errorMessage: 'Bad Request' })} />,
    );

    expect(wrapper.find('.textError')).toHaveText('Bad Request');
  });
});
