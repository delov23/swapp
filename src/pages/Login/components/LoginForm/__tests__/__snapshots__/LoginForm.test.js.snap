// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should match snapshot 1`] = `
<div
  className="formWrapper"
>
  <div
    className="formContent"
  >
    <h1
      className="formHeading"
    >
      SWAPP
    </h1>
    <form
      className="loginFormLight"
      onSubmit={[Function]}
    >
      <input
        className="inputFieldLight"
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
      <input
        className="inputFieldLight"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
      <Button
        type="submit"
      >
        Login
      </Button>
    </form>
  </div>
</div>
`;
